### PR TITLE
Fix tzinfo bugs

### PR DESCRIPTION
**Pull request by nirvdrum**

### CODE REVIEW

**Comment by nirvdrum** [comment]
I didn't add a spec for $10 and higher because any cut-off point seems arbitrary. How high should I go?

**Comment by nirvdrum** [comment]
Leap second spec added in c56b0ebaf5a22b4baa13ca9c0fb77e9b3e57b97c.

**Comment by eregon** [comment]
> I didn't add a spec for $10 and higher because any cut-off point seems arbitrary. How high should I go?

It doesn't really matter, the idea is to test for more than 1 digit so $10 or $11 is fine for example.

File: spec/ruby/core/time/shared/gm.rb [hunk 1]
```diff
@@ -26,4 +26,28 @@
     time.usec.should == 0
     time.nsec.should == 999
   end
+
+  it "handles real leap seconds" do
+    with_timezone 'right/UTC' do
```
Comment (eregon) [inline spec/ruby/core/time/shared/gm.rb]: Does a timezone have any effect on a Time created with Time.utc/Time.gm?
Also, does the timezone `right/UTC` exist?

Comment (nirvdrum) [reply spec/ruby/core/time/shared/gm.rb]: We're wading into territory I can't claim to really understand. But I derived the info from https://bugs.ruby-lang.org/issues/8885

**Pull request merged**